File: DSA/SakshamDwivedi_590013147/Day32/D32P2.py
#Solution Function
def non_repeating(arr):
    current_xor = 0
    for i in range(len(arr)):
        current_xor = current_xor ^ arr[i]
    return current_xor

#Fixed Input
print(non_repeating([2,3,2]))

#Custom Input
arr = list(int(x) for x in input("Enter the array : ").split())
print(non_repeating(arr))

File: DSA/SakshamDwivedi_590013147/Day32/D32P2.c
#include <stdio.h>

int size;
int arr[100];
int current_xor = 0;

int main(){
  printf("Enter number of elements : ");
  scanf("%d", &size);
  printf("Enter the array : ");
  for(int i = 0; i < size; i++){
    scanf("%d", &arr[i]);
  }
  for(int i = 0; i < size; i++){
    current_xor = current_xor ^ arr[i];
  }
  printf("Non Repeating : %d", current_xor);

  return 0;
}
